### PR TITLE
Fixed planning sync when changing from driver to car and vice-versa

### PR DESCRIPTION
**Pull request by Scttpr**
Checklist

- [x] I've added a mention in CHANGELOG.md

### CODE REVIEW

**Comment by Scttpr** [comment]
[conserver la bonne semaine quand je switch chauffeur <-> vehicule](https://trello.com/c/a5SxPgUJ/277-conserver-la-bonne-semaine-quand-je-switch-chauffeur-vehicule)

File: components/planning/cars/calendar.vue [hunk 1]
```diff
@@ -114,10 +116,11 @@ export default {
       this.$emit('open-edit', content);
     },
     viewChange({ startDate, endDate }) {
-      this.$emit('view-change', {
-        startDate,
-        endDate,
-      });
+      const current = new Date(startDate);
```
Comment (Nainterceptor) [inline components/planning/cars/calendar.vue]: You can probably use Luxon here to add days to date, to ensure that everything is right when we add 5 days to 30 december as an example (sometimes, Date don't change year/month when manipulating values)

File: components/planning/drivers/calendar.vue [hunk 2]
```diff
@@ -115,10 +118,11 @@ export default {
       this.$emit('open-edit', content);
     },
     viewChange({ startDate, endDate }) {
-      this.$emit('view-change', {
-        startDate,
-        endDate,
-      });
+      const current = new Date(startDate);
```
Comment (Nainterceptor) [inline components/planning/drivers/calendar.vue]: Same thing here

File: store/planning.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+export const mutations = {
+  setSelectedDate: (s, date) => {
+    s.selectedDate = date;
+  },
+};
+
+export const getters = {
+  getSelectedDate: (s) => s.selectedDate,
+};
```
Comment (Nainterceptor) [inline store/planning.js]: Please use query param for selected date

**Pull request merged**